### PR TITLE
Show error message when there are extra flags for Delete Client/Property

### LINKED ISSUE #91: Delete operation does not detect error if additional flags is included
image
image

The image shown is an example where additional flags is inputted but delete operation does not flag error. The error will also not detect error if flags such as "n/" is added.

### PR DESCRIPTION
**Pull request by FeliciaBeatrice**
Fixes #91

### CODE REVIEW

File: src/main/java/seedu/duke/parsermanager/ParseDeleteClient.java [hunk 1]
```diff
@@ -21,14 +22,18 @@ public class ParseDeleteClient extends Parser {
     private final String commandDescription;
     private final ClientList clientList;
 
+
+    private static final int CORRECT_FLAG_POSITION = 0;
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/parsermanager/ParseDeleteClient.java:26]: This is a good use of meaningful names to indicate the correct flag position. Good job :)

File: src/main/java/seedu/duke/parsermanager/ParseDeleteClient.java [hunk 2]
```diff
@@ -50,19 +55,28 @@ public Command parseCommand() throws InvalidIndexException, MissingFlagException
             throw new NotIntegerException(MESSAGE_NOT_INTEGER);
         } catch (EmptyDetailException e) {
             throw new EmptyDetailException(MESSAGE_DELETE_CLIENT_WRONG_FORMAT);
+        } catch (ExtraFlagsException e) {
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/parsermanager/ParseDeleteClient.java:58]: I really like the addition of this exception. 👍🏼

File: src/main/java/seedu/duke/parsermanager/ParseDeleteClient.java [hunk 3]
```diff
@@ -50,19 +55,28 @@ public Command parseCommand() throws InvalidIndexException, MissingFlagException
...
         String[] flags = DELETE_CLIENT_FLAGS;
         int[] flagIndexPositions = getFlagIndexPositions(rawCommandDetail, flags);
+        checkForExtraFlags(flagIndexPositions);
         checkForMissingFlags(flagIndexPositions);
         checkFlagsOrder(flagIndexPositions);
         return extractCommandDetails(rawCommandDetail, flags, flagIndexPositions);
     }
 
+    private void checkForExtraFlags(int[] flagIndexPositions) throws ExtraFlagsException {
+        if (flagIndexPositions[0] != CORRECT_FLAG_POSITION) {
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/parsermanager/ParseDeleteClient.java:75]: Perhaps you can consider replace '0' with another name?
Would it also be more readable if the entire boolean expression is replaced with a boolean variable with meaningful name?

**Review by wilsonngja — COMMENTED** [review]
Good job on the addition. It would be even better if you can include boolean variable for the conditional statements portion. Otherwise, it's still a job well done and it LGTM. 👍🏼

**Review by ngdeqi — APPROVED** [review]
Other than using 0 to access arraylist elements, no code quality violations.

**Pull request merged**